lib/miq_expression/field.rb

```diff
@@ -40,7 +40,7 @@ def valid?
   end
 
   def attribute_supported_by_sql?
-    !custom_attribute_column? && target.attribute_supported_by_sql?(column)
+    !custom_attribute_column? && target.attribute_supported_by_sql?(column) && !virtual_reflection?
```

Comment:
Does the arel-backed virtual columns feature not work at all across virtual associations, I take it?

Reply:
it is the virtual relation that gets us. if we can't get to the relation via sql, then we can't get to the column